TreeTracker/UI/Views/Messaging/Survey/SurveyViewModel.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  SurveyViewModel.swift
+//  TreeTracker
+//
+//  Created by Frédéric Helfer on 01/06/23.
+//  Copyright © 2023 Greenstand. All rights reserved.
+//
+
+import Foundation
+import Treetracker_Core
+
+protocol SurveyViewModelCoordinatorDelegate: AnyObject {
+    func surveyViewModel(_ surveyViewModel: SurveyViewModel, showNextQuestion survey: SurveyViewModel.Survey, planter: Planter, questionIndex: Int)
+    func surveyViewModel(_ surveyViewModel: SurveyViewModel, didFinishSurvey survey: SurveyViewModel.Survey)
+}
+
+protocol SurveyViewModelViewDelegate: AnyObject {
+    func surveyViewModel(_ surveyViewModel: SurveyViewModel, updateViewWith survey: SurveyViewModel.Survey)
+}
+
+class SurveyViewModel {
+
+    weak var coordinatorDelegate: SurveyViewModelCoordinatorDelegate?
+    weak var viewDelegate: SurveyViewModelViewDelegate?
+
+    private let planter: Planter
+    private var survey: SurveyViewModel.Survey
+    private let messagingService: MessagingService
+
+    init(planter: Planter, survey: SurveyViewModel.Survey, messagingService: MessagingService, questionIndex: Int) {
+        self.planter = planter
+        self.survey = survey
+        self.messagingService = messagingService
+        self.questionIndex = questionIndex
+    }
+
+    var questionIndex: Int
+
+    var title: String {
+        let title = survey.response ? L10n.Survey.Title.response : L10n.Survey.Title.question
+        return "\(title) \(questionIndex + 1)/\(survey.questions.count)"
+    }
+
+    var numberOfRowsInSection: Int {
```

Comment:
I think just `numberOfChoices` would be better here.

The viewmodel doesn't really need knowledge that a table/collection view is part of the view so no need to mimic the function names.

Also the variable name includes `inSection` which implies you would be passing section as a parameter which we're not.

Reply:
Great! I'm always learning to make the code cleaner with these reviews!